x-pack/plugins/security_solution/docs/testing/test_plans/detection_response/prebuilt_rules/upgrade_review_algorithms.md

```diff
@@ -288,3 +350,43 @@ Examples:
 | algorithm        | field_name | base_version | current_version        | target_version         | merged_version                 |
 | array of scalars | tags       | N/A          | ["one", "two", "four"] | ["one", "two", "five"] | ["one", "two", "four", "five"] |
 ```
+
+#### **Scenario: `-AB` - Rule field is a solvable `data_source` object**
+
+**Automation**: 1 integration test with mock rules + a set of unit tests for the algorithm
+
+```Gherkin
+Given at least 1 installed prebuilt rule has a new version available
+And the base version of the rule cannot be determined
+And customized data_source field is different than the Elastic update in this upgrade (current version != target version)
+And current version and target version are both array fields in data_source
+Then for data_source field the diff algorithm should output a custom merged version with a solvable conflict
+And arrays should be deduplicated before comparison
+And arrays should be compared sensitive of case
+And arrays should be compared agnostic of order
+And data_source field should be returned from the `upgrade/_review` API endpoint
+And data_source field should be shown in the upgrade preview UI
+
+Examples:
+| algorithm    | base_version | current_version                                                     | target_version                                                     | merged_version                                                              |
+| data_source  | N/A          | {type: "index_patterns", "index_patterns": ["one", "two", "three"]} | {type: "index_patterns", "index_patterns": ["one", "two", "four"]} | {type: "index_patterns", "index_patterns": ["one", "two", "three", "four"]} |
+
+```
+
+#### **Scenario: `-AB` - Rule field is a non-solvable `data_source` object**
+
+**Automation**: 1 integration test with mock rules + a set of unit tests for the algorithm
+
+```Gherkin
+Given at least 1 installed prebuilt rule has a new version available
+And the base version of the rule cannot be determined
+And customized data_source field is different than the Elastic update in this upgrade (current version != target version)
+And current version and target version are not both array fields in data_source
+Then for data_source field the diff algorithm should output the target version as the merged version with a non-solvable conflict
```

Comment:
For this scenario, we consider them `SOLVABLE` conflicts by definition, right?

Reply:
Yep, good catch